### PR TITLE
Feature/mob 4390 checkout price bottom view

### PR DESCRIPTION
**Pull request by BartlomiejSopalaKarhoo**
## JIRA
[MOB-4390](https://karhoo.atlassian.net/browse/MOB-4390)

## DESCRIPTION
This PR contains origin/MOB-4441-price-details-bottom-sheet branch, so please review and merge Diana's branch at the beginning 

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

[MOB-4390]: https://karhoo.atlassian.net/browse/MOB-4390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/DetailsCellView.swift [hunk 1]
```diff
@@ -61,13 +61,14 @@ struct DetailsCellView: View {
...
+            delegate?.onTap() // will be removed when Old Checkout will be removed
+            viewModel.onTap()
         })
     }
 }
 
 struct DetailsViewCellPreviews: PreviewProvider {
     static var previews: some View {
-        DetailsCellView(model: DetailsCellViewModel(title: "Passanger", subtitle: "add details", iconName: "kh_uisdk_passenger"), delegate: nil)
+        DetailsCellView(viewModel: DetailsCellViewModel(title: "Passanger", subtitle: "add details, this is excample of text containing more than 2, or even more than 3 lines. We need that kind of text for comments cell, in this case input for user is unlimited. ", iconName: "kh_uisdk_passenger", onTap: {print("tapped")}), delegate: nil)
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/DetailsCellView.swift:72]: It's more common to insert a `Lorem ipsum` text in the future to mimic long texts. Can leave this as is, just a note on future long placeholder needs.

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                HStack(spacing: UIConstants.Spacing.small) {
+                    KarhooAsyncImage(urlString: viewModel.fleetIconUrl)
+                        .frame(
+                            maxWidth: UIConstants.Dimension.Icon.medium,
+                            maxHeight: UIConstants.Dimension.Icon.medium
+                        ).scaledToFill()
+                        
+                    Text(viewModel.fleetName)
+                        .font(Font(KarhooUI.fonts.captionSemibold()))
+                        .foregroundColor(KarhooUI.colors.text.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:36]: Can use `Color(KarhooUI.colors.text`)

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    
+    var body: some View {
+        HStack(alignment: .center, spacing: UIConstants.Spacing.small) {
+            KarhooAsyncImage(urlString: viewModel.carIconUrl)
+                .frame(width: 80, height: 80)
+                .scaledToFit()
+            VStack(alignment: .leading, spacing: UIConstants.Spacing.xSmall) {
+                Text(viewModel.title)
+                    .font(Font(KarhooUI.fonts.headerSemibold()))
+                    .foregroundColor(KarhooUI.colors.text.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:22]: Can use `Color(KarhooUI.colors.text)`

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                    KarhooAsyncImage(urlString: viewModel.fleetIconUrl)
+                        .frame(
+                            maxWidth: UIConstants.Dimension.Icon.medium,
+                            maxHeight: UIConstants.Dimension.Icon.medium
+                        ).scaledToFill()
+                        
+                    Text(viewModel.fleetName)
+                        .font(Font(KarhooUI.fonts.captionSemibold()))
+                        .foregroundColor(KarhooUI.colors.text.getColor())
+
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:37]: Unnecessary extra lines

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                        .foregroundColor(KarhooUI.colors.text.getColor())
+                    
+                    
+                }
+                
+            }
+            Spacer()
+            
+                
+        }.padding(UIConstants.Spacing.standard)
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:45]: .padding on new line

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            HStack {
+                Group {
+                    Image(uiImage: .uisdkImage(iconName))
+                        .frame(
+                            width: UIConstants.Dimension.Icon.small,
+                            height: UIConstants.Dimension.Icon.small
+                        )
+                    Text("\(value)")
+                        .font(Font(KarhooUI.fonts.footnoteSemiold()))
+                        .foregroundColor(KarhooUI.colors.text.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:97]: Can use `Color(KarhooUI.colors.text)`

File: KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                }
+            }
+            .padding(UIConstants.Spacing.xSmall)
+            .background(
+                RoundedRectangle(cornerRadius: UIConstants.CornerRadius.xSmall)
+                    .fill(KarhooUI.colors.background1.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/Checkout/VehicleDetailsCard.swift:103]: Can use `Color(KarhooUI.colors.background1)`

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift [hunk 8]
```diff
@@ -16,15 +16,50 @@ struct NewCheckoutView: View {
...
+
+        ZStack {
+            ScrollView(showsIndicators: false) {
+                VStack(spacing: 0) {
+                    VStack(spacing: 0) {
+                        dateView
+                        addressView
+                    }
+                    .padding(.horizontal, UIConstants.Spacing.standard)
+                    .background(KarhooUI.colors.background2.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift:28]: Can use `Color(KarhooUI.colors.background2)`

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift [hunk 9]
```diff
@@ -35,6 +70,7 @@ struct NewCheckoutView: View {
                 .fixedSize()
             Spacer()
         }.padding(.top, UIConstants.Spacing.standard)
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift:72]: Formatting issue. `.padding` should be on a new line

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift [hunk 10]
```diff
@@ -35,6 +70,7 @@ struct NewCheckoutView: View {
...
+        .background(KarhooUI.colors.background2.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift:73]: Can use `Color((KarhooUI.colors.background2)`

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift [hunk 11]
```diff
@@ -53,4 +89,45 @@ struct NewCheckoutView: View {
...
+            VStack(spacing: 0){
+                Spacer()
+                Color(KarhooUI.colors.border)
+                    .frame(height: UIConstants.Dimension.Border.standardWidth)
+                HStack(alignment: .top, spacing: UIConstants.Spacing.standard) {
+                    
+                    VStack(alignment: .leading, spacing: 0){
+                        Text(viewModel.quote.quoteType.description)
+                            .font(Font(KarhooUI.fonts.captionBold()))
+                            .foregroundColor(KarhooUI.colors.textLabel.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift:105]: Can use `Color(KarhooUI.colors.textLabel)`

File: KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift [hunk 12]
```diff
@@ -53,4 +89,45 @@ struct NewCheckoutView: View {
...
+                        HStack(spacing: UIConstants.Spacing.xSmall){
+                            Text(CurrencyCodeConverter.toPriceString(quote: viewModel.quote))
+                                .font(Font(KarhooUI.fonts.title2Bold()))
+                                .foregroundColor(KarhooUI.colors.text.getColor())
```
Comment (dianakarh) [inline KarhooUISDK/Screens/CheckoutScreen/NewCheckout/NewCheckoutView.swift:109]: Can use `Color(KarhooUI.colors.text)`

**Review by AleksanderWedrychowskiKarhoo — APPROVED** [review]

**Pull request merged**